crates/cmd/src/report.rs

```diff
@@ -0,0 +1,206 @@
+use clap::ValueEnum;
+use std::path::PathBuf;
+use std::process::{ExitCode, Termination};
+use wardstone::key;
+use wardstone::primitive::asymmetric::Asymmetric;
+use wardstone_core::primitive::hash::Hash;
+
+use crate::Verbosity;
+
+const INDENT: &str = "         ";
+
+pub(crate) enum GoodPath {
+  Hash(Hash, Hash),
+  SigAlg(Asymmetric, Asymmetric),
+}
+
+impl GoodPath {
+  fn display(&self, format: ReportFormat) -> String {
+    match format {
+      ReportFormat::Human => match self {
+        GoodPath::Hash(got, want) => {
+          format!("{INDENT}OK hash function(got: {got}, want: {want})")
+        },
+        GoodPath::SigAlg(got, want) => {
+          format!("{INDENT}OK signature algorithm(got: {got}, want: {want})")
+        },
+      },
+      ReportFormat::Json => match self {
+        GoodPath::Hash(got, want) => {
+          format!(r#""hash_function": {{ "passed": true, "got": "{got}", "want": "{want}" }}"#)
+        },
+        GoodPath::SigAlg(got, want) => {
+          format!(
+            r#""signature_algorithm": {{ "passed": true, "got": "{got}", "want": "{want}" }}"#
+          )
+        },
+      },
+    }
+  }
+}
+
+pub(crate) enum BadPath {
+  ReadError(key::Error),
+  MismatchedHash(Hash, Hash),
+  MismatchedSigAlg(Asymmetric, Asymmetric),
+}
+
+impl BadPath {
+  fn display(&self, form: ReportFormat) -> String {
+    match form {
+      ReportFormat::Human => match self {
+        BadPath::ReadError(e) => format!("{e}"),
+        BadPath::MismatchedHash(got, want) => {
+          format!("{INDENT}FAILED hash function(got: {got}, want: {want})")
+        },
+        BadPath::MismatchedSigAlg(got, want) => {
+          format!("{INDENT}FAILED signature algorithm(got: {got}, want: {want})")
+        },
+      },
+      ReportFormat::Json => match self {
+        BadPath::ReadError(e) => format!(r#"{INDENT}"error": "{e}""#),
+        BadPath::MismatchedHash(got, want) => {
+          format!(r#""hash_function": {{ "passed": false, "got": "{got}", "want": "{want}" }}"#)
+        },
+        BadPath::MismatchedSigAlg(got, want) => {
+          format!(
+            r#""signature_algorithm": {{ "passed": false, "got": "{got}", "want": "{want}" }}"#
+          )
+        },
+      },
+    }
+  }
+}
+
+type CheckResult = Result<GoodPath, BadPath>;
```

Comment:
See comment that follows.

```suggestion
type AuditResult = Result<Success, Failure>;
```